build/android/src/main/java/net.minetest.minetest/TextEntryActivity.java

```diff
@@ -0,0 +1,88 @@
+package net.minetest.minetest;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.InputType;
+import android.view.KeyEvent;
+import android.view.View;
+import android.view.View.OnKeyListener;
+import android.widget.EditText;
+
+public class TextEntryActivity extends Activity {
+	public AlertDialog inputDialog;
+	public EditText inputWidget;
+
+	private final int INPUT_MULTILINE = 0;
```

Comment:
My understanding is that emuns are basically complete classes in java just with different syntax, and anyways the values have to be consistent between the C++ and Java source so it's better to be explicit.